bug(analytics): AdvancedAnalytics v-for idx typed as string|number causes TS2365 errors

## Problem

`vue-tsc` reports three `TS2365` errors in `AdvancedAnalytics.vue`:

```
src/components/analytics/AdvancedAnalytics.vue(249,61): error TS2365: Operator '+' cannot be applied to types 'string | number' and 'number'.
src/components/analytics/AdvancedAnalytics.vue(249,74): error TS2365: Operator '+' cannot be applied to types 'string | number' and 'number'.
src/components/analytics/AdvancedAnalytics.vue(278,40): error TS2365: Operator '+' cannot be applied to types 'string | number' and 'number'.
```

## Root Cause

The `v-for` index variable (`idx`) over `engagementMetrics.feature_popularity` and `usageHeatmap.peak_hours` is inferred as `string | number` because the collection type is not narrowly typed as an array. The template then tries `idx + 1` which TS rejects.

Lines 249/278 in the compiled output correspond to:
- `'rank-' + (idx + 1)` — class binding on feature popularity table
- `{{ idx + 1 }}` — peak hours rank display

## Fix

Either:
- Type `feature_popularity` and `peak_hours` as proper arrays in the data interface so `idx` infers as `number`, OR
- Cast at use site: `(idx as number) + 1`

## Acceptance Criteria
- [ ] `vue-tsc --noEmit -p tsconfig.app.json` reports no TS2365 errors in `AdvancedAnalytics.vue`

--- a/autobot-frontend/src/components/analytics/AdvancedAnalytics.vue
+++ b/autobot-frontend/src/components/analytics/AdvancedAnalytics.vue
@@ -244,9 +244,9 @@
             </tr>
           </thead>
           <tbody>
-            <tr v-for="(feature, idx) in engagementMetrics.feature_popularity" :key="feature.feature">
+            <tr v-for="(feature, idx) in (engagementMetrics.feature_popularity as any[])" :key="feature.feature">
               <td>
-                <span class="rank-badge" :class="'rank-' + (idx + 1)">{{ idx + 1 }}</span>
+                <span class="rank-badge" :class="'rank-' + ((idx as number) + 1)">{{ (idx as number) + 1 }}</span>
                 {{ feature.feature }}
               </td>
               <td class="text-right">{{ formatNumber(feature.views) }}</td>
@@ -271,11 +271,11 @@
         </template>
         <div class="peak-hours-list">
           <div
-            v-for="(peak, idx) in usageHeatmap.peak_hours"
+            v-for="(peak, idx) in (usageHeatmap.peak_hours as any[])"
             :key="peak.hour"
             class="peak-hour-item"
           >
-            <span class="peak-rank">{{ idx + 1 }}</span>
+            <span class="peak-rank">{{ (idx as number) + 1 }}</span>
             <span class="peak-time">{{ peak.hour }}:00</span>
             <span class="peak-events">{{ formatNumber(peak.total_events) }} {{ $t('analytics.advanced.events') }}</span>
           </div>

